frontend/app/services/work-package-service.js

```diff
@@ -75,6 +76,47 @@ module.exports = function($http,
   }
 
   var WorkPackageService = {
+    initializeWorkPackage: function(projectIdentifier, initialData) {
+      var changes = _.clone(initialData);
+      if (initialData.type) {
+        changes._links = changes.links || {};
+        changes._links.type = {
+          href: initialData.type
+        };
+        delete changes.type;
+      }
+      var wp = {
+        embedded: {},
+        props: {},
+        links: {
+          update: HALAPIResource
+            .setup(PathHelper
+              .projectWorkPackagesFormPath(projectIdentifier)),
+          updateImmediately: HALAPIResource.setupLink(
+            PathHelper.projectWorkPackagesPath(projectIdentifier),
+            { method: 'post' }
+          )
+        }
+      };
+      var options = { ajax: {
+          method: 'POST',
+          headers: {
+            Accept: 'application/hal+json'
+          },
+          data: JSON.stringify(changes),
+          contentType: 'application/json; charset=utf-8'
+        }};
```

Comment:
A note on dryness:

I can't imagine this is the first time we define those options. [They](https://github.com/NobodysNightmare/openproject/blob/2995ad8f53af0c06755d31554f0d3c28e8055baa/frontend/app/services/work-package-service.js#L104-104) [look](https://github.com/NobodysNightmare/openproject/blob/2995ad8f53af0c06755d31554f0d3c28e8055baa/frontend/app/services/work-package-service.js#L227-227) [pretty](https://github.com/NobodysNightmare/openproject/blob/2995ad8f53af0c06755d31554f0d3c28e8055baa/frontend/app/services/work-package-service.js#L255-255) [default](https://github.com/NobodysNightmare/openproject/blob/2995ad8f53af0c06755d31554f0d3c28e8055baa/frontend/app/services/work-package-service.js#L270-270).